File: java-solid-principle-sample-code/src/com/principle/solid/liskov_substitution/NoShape.java
package com.principle.solid.liskov_substitution;

public class NoShape implements Shape{

	/** 
		Liskov Substitution Principle violation - It is Substitution able, Even if we don't override.
        We have to either provide Interface `Default` implementation /or/ `Abstract` Class for Substitution
	 */
    @Override
    public double area() {
        throw new IllegalStateException("Cannot calculate area");
    }
}

File: python-solid-principle-sample-code/com/principle/solid/liskov_substitution/NoShape.py
from Shape import Shape
from IllegalStateException import IllegalStateException


class NoShape(Shape):

    def __init__(self):
        pass

    """
        Liskov Substitution Principle violation - It is Substitution able, Even if we don't override.
        We have to either provide Interface `Default` implementation /or/ `Abstract` Class for Substitution
    """
    def area(self):
        raise IllegalStateException("Cannot calculate area")
